### PR TITLE
Issue #3273219 by vnech: Make it possible for users to receive the email after enrolling in the event

### PR DESCRIPTION
**Pull request by nechai**
## Problem
As a user I should receive an email confirmation when I enroll in an event.

## Solution
Add a new message template responsible for the messaging user

## Issue tracker
- https://www.drupal.org/project/social/issues/3273219
- https://getopensocial.atlassian.net/browse/YANG-7131

## Theme issue tracker
- https://git.drupalcode.org/project/socialbase/-/merge_requests/115
- https://git.drupalcode.org/project/socialbase/-/merge_requests/128
- https://git.drupalcode.org/project/socialblue/-/merge_requests/75

## How to test
**Before testing:**
- [x] For testing is useful to disable the module `social_advanced_queue` as it can blocks emails sending
- [x] Also make sure the option `activity_send_offline_window` in `activity_send.settings` is set to `0`

**STR**:
- [x] Login as SM
- [x] Create an event
- [x] Make sure the event has open enrolling method
- [x] During event creation enable the option “Send confirmation email when a member enrols in a event“
- [x] Login as LU
- [x] Enroll to event
- [x] Login as an admin
- [x] Run cron (2 times)
- [x] Check if LU received the email

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] If applicable (I.E. new hook_updates) the update path from previous versions (major and minor versions) are tested
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
Screenshot 2022-04-13 at 00 04 04

## Release notes
*Make possible for user to receive the email after enrolling to the event.*

## Change Record

- Introduced two new templates for `event` node type:
  - [event-location-address.html.twig](https://github.com/goalgorilla/open_social/pull/2853/files#diff-094a550ebd83fbd134e8e745a8476ff6f4d838548b2b273fbe45157e8af419c8)
  - [event-date-advanced.html.twig](https://github.com/goalgorilla/open_social/pull/2853/files#diff-3419b020eb1e8f210882e8087d8f79e1697c3238cf27a6a1a00cfcf9fce99d54)
- Introduced new node display view mode that can be used for displaying on emails:
  - [core.entity_view_mode.node.email_card.yml](https://github.com/goalgorilla/open_social/pull/2853/files#diff-3677555b70938e8e8b650c97d7e801ec55ab22c0c92e5d18c7dbe8306ac01a4d)
- Added new activity context:
  - [OwnerEventEnrollmentActivityContext.php](https://github.com/goalgorilla/open_social/pull/2853/files#diff-6d081eb513a0728440fe72ef33ce5c77ab8f9663567720e9fa3f0fe53e630248)
- Added new activity entity condition:
  - [EventEnrollmentStandaloneEnrollActivityEntityCondition.php](https://github.com/goalgorilla/open_social/pull/2853/files#diff-c57f2a7bcaf532ebef09f98bc33ff307476ace262d0562c798f4abbf78a32cbf)

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

File: modules/custom/activity_logger/activity_logger.module [hunk 1]
```diff
@@ -108,7 +108,7 @@ function activity_logger_form_message_template_form_alter(&$form, FormStateInter
     '#title' => t('The entity condition that are affected for this message'),
     '#description' => t('Select a entity condition for this message.'),
     '#default_value' => $activity_entity_condition_value,
-    '#options' => $activity_entity_condition_options,
+    '#options' => ['' => '- none -'] + $activity_entity_condition_options,
```
Comment (rollki) [inline modules/custom/activity_logger/activity_logger.module:111]: Can we use `'#empty_option' => t('- none -'),` instead?

File: modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - field.storage.node.field_event_send_confirmation
+    - node.type.event
+  enforced:
+    module:
+      - social_event
+id: node.event.field_event_send_confirmation
+field_name: field_event_send_confirmation
+entity_type: node
+bundle: event
+label: 'Send confirmation after standalone enrolment'
```
Comment (rollki) [inline modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml:14]: Does this label correct?

File: modules/social_features/social_event/config/optional/core.entity_view_display.node.event.email_card.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+mode: email_card
+content:
+  field_event_address:
+    type: address_plain
+    weight: 1
+    label: hidden
+    settings: {  }
+    third_party_settings: {  }
+    region: content
+  field_event_addtocal:
```
Comment (rollki) [inline modules/social_features/social_event/config/optional/core.entity_view_display.node.event.email_card.yml:29]: It's an optional submodule, so not sure if this `field_event_addtocal` should be in this config.

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    activity_bundle_entities:
+      event_enrollment-event_enrollment: event_enrollment-event_enrollment
+    activity_action: create_entitiy_action
+    activity_context: owner_event_enrollment_activity_context
+    activity_destinations:
+      email: email
+    activity_create_direct: true
+    activity_aggregate: false
+    activity_entity_condition: event_enrollment_standalone_enroll
+    email_subject: 'You were enrolled to an event'
```
Comment (rollki) [inline modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:21]: Does this subject confirmed by Jessica?

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+template: user_was_enrolled_to_event
+label: 'A user was enrolled to an Event'
+description: 'A user was enrolled to an Event'
```
Comment (rollki) [inline modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:24]: Not sure, but maybe `A user has enrolled in an Event`?

File: modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11100.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+mode: email_card
+content:
+  field_event_address:
+    type: address_plain
+    weight: 1
+    label: hidden
+    settings: {  }
+    third_party_settings: {  }
+    region: content
+  field_event_addtocal:
```
Comment (rollki) [inline modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11100.yml:29]: It's an optional submodule, so not sure if this field_event_addtocal should be in this config.

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml (hunk 4 above)
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml]: @rollki The subject was corrected by Jessica and changed in the PR.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml (hunk 5 above)
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml]: @rollki changed with suggestions provided by Jessica.

File: modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml (hunk 2 above)
Comment (nechai) [reply modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml:14]: @rollki I added readable label for the field more-less described what it does. But in the form users will see description of the field (see [here](https://github.com/goalgorilla/open_social/pull/2853/files#diff-4c305b27578c9c82905e13f0df9785fa52e1ec815883ca4fabf05b624fda0f8cR745))

File: modules/custom/activity_logger/activity_logger.module (hunk 1 above)
Comment (nechai) [reply modules/custom/activity_logger/activity_logger.module:111]: @rollki Fixed

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by rollki** [comment]
Good job, please only check if we can dynamically add `field_event_addtocal` and if `#empty_option` works properly.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

File: modules/social_features/social_event/config/optional/core.entity_view_display.node.event.email_card.yml (hunk 3 above)
Comment (nechai) [reply modules/social_features/social_event/config/optional/core.entity_view_display.node.event.email_card.yml]: @rollki I removed this field from view display. Even more, if we take a look to [this function](https://github.com/goalgorilla/open_social/blob/main/modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module#L32) we can see that the field `field_event_addtocal` can't be managed in any view displays 🤷

File: modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11100.yml (hunk 6 above)
Comment (nechai) [reply modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11100.yml]: @rollki I removed this field from view display. Even more, if we take a look to [this function](https://github.com/goalgorilla/open_social/blob/main/modules/social_features/social_event/modules/social_event_addtocal/social_event_addtocal.module#L32) we can see that the field `field_event_addtocal` can't be managed in any view displays 🤷

**Comment by rollki** [comment]
LGTM, I have restarted failed tests.
@Ressinel Could you please take a look at this PR?

**Review by rollki — APPROVED** [review]

**Comment by Ressinel** [comment]
Hi @nechai @rollki 

I saw that behat tests failed again and tried to restart them again. Unfortunately, they failed even after my restart. It seems that these errors relate to these changes

**Comment by Ressinel** [comment]
@nechai Could you add a milestone, please?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2853-ue6dzqsfn2aztn40u6htifo6sox9f3bh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/624a29c8b1a46a5aa1d380de

**Comment by tbsiqueira** [comment]
Hi @nechai, can you please resolve the conflict so we can follow up with the tests and merge? Thank you!

**Comment by tbsiqueira** [comment]
Hi @nechai, tests 3 and 5 are failing at the same place for a couple of times, can you please take a look at them? 

https://app.travis-ci.com/github/goalgorilla/open_social/jobs/570288450
https://app.travis-ci.com/github/goalgorilla/open_social/jobs/570288452

**Comment by nechai** [comment]
Hi @tbsiqueira 
Both tests are fixed.

**Review by Ressinel — CHANGES_REQUESTED** [review]

File: modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+description: 'Send confirmation email when a member enrols in a event'
```
Comment (Ressinel) [inline modules/social_features/social_event/config/install/field.field.node.event.field_event_send_confirmation.yml:15]:
```suggestion
description: 'Send a confirmation email when a member enrolls in an event'
```

File: modules/social_features/social_event/config/static/field.field.node.event.field_event_send_confirmation_11301.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    - node.type.event
+  enforced:
+    module:
+      - social_event
+id: node.event.field_event_send_confirmation
+field_name: field_event_send_confirmation
+entity_type: node
+bundle: event
+label: 'Send confirmation after standalone enrolment'
+description: 'Send confirmation email when a member enrols in a event'
```
Comment (Ressinel) [inline modules/social_features/social_event/config/static/field.field.node.event.field_event_send_confirmation_11301.yml:15]:
```suggestion
description: 'Send a confirmation email when a member enrolls in an event'
```

**Review by Ressinel — CHANGES_REQUESTED** [review]

**Comment by Ressinel** [comment]
@nechai I can't receive an email after enrolling in the event. Here are steps to check:
- Create event by user_1 with the setting `Send confirmation email when a member enrols in a event` enabled
- By user_2 enroll in the created event, he won't receive an email

File: modules/social_features/social_core/social_core.install [hunk 9]
```diff
@@ -1560,3 +1561,13 @@ function social_core_update_11301(): void {
     $config->save();
   }
 }
+
+/**
+ * Add a new view mode for nodes - Email card.
+ */
+function social_core_update_11302(): void {
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_core/social_core.install]: Fixed

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    activity_bundle_entities:
+      event_enrollment-event_enrollment: event_enrollment-event_enrollment
+    activity_action: create_entitiy_action
+    activity_context: owner_event_enrollment_activity_context
+    activity_destinations:
+      email: email
+    activity_create_direct: true
+    activity_aggregate: false
+    activity_entity_condition: event_enrollment_standalone_enroll
+    email_subject: 'You’ve enrolled for an event'
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:21]: Fixed

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+template: user_was_enrolled_to_event
+label: 'A user was enrolled for the event'
+description: 'A user was enrolled for the event'
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:24]: Fixed

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    activity_action: create_entitiy_action
+    activity_context: owner_event_enrollment_activity_context
+    activity_destinations:
+      email: email
+    activity_create_direct: true
+    activity_aggregate: false
+    activity_entity_condition: event_enrollment_standalone_enroll
+    email_subject: 'You’ve enrolled for an event'
+template: user_was_enrolled_to_event
+label: 'A user was enrolled for the event'
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:23]: Fixed

File: modules/social_features/social_event/social_event.install [hunk 13]
```diff
@@ -2107,3 +2107,35 @@ function social_event_update_10311(): string {
   // Output logged messages to related channel of update execution.
   return $updateHelper->logger()->output();
 }
+
+/**
+ * Add new field to Event node type; Add new message template.
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/social_event.install:2112]: Fixed

File: modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11301.yml [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+mode: email_card
+content:
+  field_event_address:
+    type: address_plain
+    weight: 1
+    label: hidden
+    settings: {  }
+    third_party_settings: {  }
+    region: content
+  field_event_addtocal:
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11301.yml]: Fixed

File: modules/social_features/social_event/src/Plugin/ActivityContext/OwnerEventEnrollmentActivityContext.php [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    return $recipients;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isValidEntity(EntityInterface $entity): bool {
+    return (bool) $entity->getEntityTypeId() == 'event_enrollment';
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/src/Plugin/ActivityContext/OwnerEventEnrollmentActivityContext.php]: Fixed

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2853-rvejatkengsxnsluo76dkqdxywrzccb3.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/628b4e3e6eb9e7df1ad7ac8c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2853-rvejatkengsxnsluo76dkqdxywrzccb3.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/628b4e3e6eb9e7df1ad7ac8c

File: modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+description: 'A user was enrolled for the event'
+text:
+  -
+    value: ''
+    format: full_html
+  -
+    value: ''
+    format: full_html
+  -
+    value: "<p>This is to confirm that you've enrolled for the event <u>[social_event:event_iam_organizing]</u>.</p>\r\n\r\n<p>[message:preview]</p>\r\n\r\n<p>[message:cta_button]</p>\r\n"
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/optional/message.template.user_was_enrolled_to_event.yml:33]: Fixed

File: modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11301.yml [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    weight: 3
+    region: content
+    settings: {  }
+    third_party_settings: {  }
+  field_event_date:
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/static/core.entity_view_display.node.event.email_card_11301.yml]: Fixed

File: modules/social_features/social_event/templates/event-location-address.html.twig [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+ */
+#}
+<div{{ attributes }}>
+  {% if icon and (location or address) %}
+    <span class="icon no-b-lazy"><img src="{{ icon }}" alt={{ 'Event location'|t }}/></span>
+  {% endif %}
+  <span class="sr-only">{{ 'Event location'|t }}</span>
+  <span class="inline-center location">
+    {{ location }}
+    {% if address is not empty and location is not empty %} • {% endif %}
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/templates/event-location-address.html.twig:27]: Unfortunately, this encoded sign doesn't work properly in emails (it won't be decoded).

File: modules/social_features/social_event/social_event.install [hunk 19]
```diff
@@ -2107,3 +2107,35 @@ function social_event_update_10311(): string {
...
+ */
+function social_event_update_11301(): void {
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/social_event.install]: Fixed

File: modules/social_features/social_event/config/install/core.entity_view_display.node.event.activity.yml [hunk 20]
```diff
@@ -104,3 +104,4 @@ hidden:
   like_and_dislike: true
   search_api_excerpt: true
   field_event_all_day: true
+  field_event_send_confirmation: true
```
[the remark this answers is not in the document]
Comment (nechai) [reply modules/social_features/social_event/config/install/core.entity_view_display.node.event.activity.yml:107]: It doesn't necessary, as new fields in view displays are always disabled by default

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr2853-rvejatkengsxnsluo76dkqdxywrzccb3.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/628b4e3e6eb9e7df1ad7ac8c

**Review by Ressinel — APPROVED** [review]
@nechai Great work🚀 
Looks good to me.

**Comment by rollki** [comment]
Patch for OS 11.3.x with correct hook numbers - https://www.drupal.org/project/social/issues/3273219#comment-14642973

**Comment by bramtenhove** [comment]
@nechai, this PR introduced an issue where notifications are no longer shown in the Notifications center.

I tracked it down to the replacement of the `renderPlain()` to the `renderRoot()` you did at https://github.com/goalgorilla/open_social/pull/2853/files#diff-384c876efe3dce8aeaa90a3c418a9677fb57b31689f4457d211fe5df6131d13cR160.

When you're trying to open such notifications in the Notifications center you will get the following message:
`A stray renderRoot() invocation is causing bubbling of attached assets to break.`

Is there a possibility to move the `renderRoot()` to later in the process? Upon actually sending the email? Because currently it'll break for everything that is not an email.
Why couldn't we use `->getContentPreview()` as it was before?

**Comment by nechai** [comment]
@bramtenhove Indeed, using `enderPlain()` solve the issue 👍
Why couldn't we use `renderRoot()` rather then `getContentPreview()`? Because as you can see in the code we need to render node in `email_card` display mode and `getContentPreview()` doesn't allows do that.

**Pull request merged**